test-clients/src/main/java/com/hedera/services/bdd/suites/contract/traceability/NewTraceabilitySuite.java

```diff
@@ -0,0 +1,171 @@
+/*
+ * Copyright (C) 2022 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hedera.services.bdd.suites.contract.traceability;
+
+import static com.hedera.services.bdd.spec.HapiApiSpec.defaultHapiSpec;
+import static com.hedera.services.bdd.spec.queries.QueryVerbs.getContractBytecode;
+import static com.hedera.services.bdd.spec.queries.QueryVerbs.getTxnRecord;
+import static com.hedera.services.bdd.spec.transactions.TxnVerbs.contractCreate;
+import static com.hedera.services.bdd.spec.transactions.TxnVerbs.cryptoCreate;
+import static com.hedera.services.bdd.spec.transactions.TxnVerbs.uploadInitCode;
+import static com.hedera.services.bdd.spec.utilops.CustomSpecAssert.allRunFor;
+import static com.hedera.services.bdd.spec.utilops.UtilVerbs.assertionsHold;
+import static com.hedera.services.bdd.spec.utilops.UtilVerbs.withOpContext;
+import static com.hedera.services.bdd.suites.contract.Utils.extractBytecodeUnhexed;
+import static com.hedera.services.bdd.suites.contract.Utils.getResourcePath;
+import static com.hederahashgraph.api.proto.java.ResponseCodeEnum.SUCCESS;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import com.google.protobuf.ByteString;
+import com.hedera.services.bdd.spec.HapiApiSpec;
+import com.hedera.services.bdd.spec.HapiSpecSetup;
+import com.hedera.services.bdd.spec.utilops.CustomSpecAssert;
+import com.hedera.services.bdd.suites.HapiApiSuite;
+import com.hedera.services.stream.proto.ContractBytecode;
+import com.hedera.services.stream.proto.TransactionSidecarRecord;
+import java.io.IOException;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import org.apache.commons.lang3.tuple.Pair;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+public class NewTraceabilitySuite extends HapiApiSuite {
+    private static final Logger log = LogManager.getLogger(NewTraceabilitySuite.class);
+
+    private static SidecarWatcher sidecarWatcher;
+    private static CompletableFuture<Void> sidecarWatcherTask;
+
+    public static void main(String... args) {
+        new NewTraceabilitySuite().runSuiteSync();
+    }
+
+    private static void initWatching() {
```

Comment:
Agreed, we shouldn't even start executing the tests if the watch service can't be initialized, because we won't be able to assert the sidecar records.

Refactored the methods in `SidecarWatcher` now, decoupling the initialization of the watch service and the actual watching for sidecars. So as it currently stands:
*  if watch service init fails -> we fail fast; 
* if an exception occurs in the watch thread (while reading sidecar files, invalid `WatchKey`, etc.) -> the `assertSidecars` test will catch that there are unmatched sidecars and will fail.